server/src/main/java/io/crate/replication/logical/action/TransportDropSubscriptionAction.java

```diff
@@ -21,130 +21,283 @@
+        );
+        return future;
+    }
+
+    /**
+     * Opens tables after removing replication setting
+     * and consequently updates DocTableInfo-s with the normal engine and makes tables writable.
+     */
+    private CompletableFuture<Void> submitOpenSubscriptionsTablesTask(List<OpenCloseTable> openCloseTables) {
+        var future = new CompletableFuture<Void>();
+
+        future.complete(null);
+
+        /* TODO: Uncomment code below once TransportCloseTable is adapted to handle multiple tables.
+        // Check if we're still the elected master before sending third update task
+        if (!clusterService.state().nodes().isLocalNodeElectedMaster()) {
+            logger.error("Couldn't execute task 'drop-sub-open-tables'. Please run command DROP SUBSCRIPTION again.");
+            future.completeExceptionally(new IllegalStateException("Master was re-elected, cannot execute 'drop-sub-open-tables'"));
+        }
+
+        clusterService.submitStateUpdateTask(
+            "drop-sub-open-tables",
+            new ClusterStateUpdateTask() {
+                @Override
+                public ClusterState execute(ClusterState currentState) {
+                    // extract relevant logic from AddCloseBlocksTask(listener, request)),
+                    // make it accept multiple tables - update builders inside loop
+                    // then call that code here and in TransportCloseTable's execute call extracted method like f(List.of(request.relation))
+                }
+
+                @Override
+                public void onFailure(String source, Exception e) {
+                    logger.error("Couldn't execute task 'drop-sub-open-tables'. Please run command DROP SUBSCRIPTION again.");
+                    future.completeExceptionally(e);
+                }
+
+                @Override
+                public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {
+                    future.complete(null);
+                }
+            }
+        );
+        */
+        return future;
+    }
+
+    private List<IndexMetadata> getSubscriptionIndices(@Nonnull String subscriptionToDrop, ImmutableOpenMap<String, IndexMetadata> indices) {
+        List<IndexMetadata> relevantIndices = new ArrayList<>();
+        Iterator<IndexMetadata> indicesIterator = indices.valuesIt();
+        while (indicesIterator.hasNext()) {
+            IndexMetadata indexMetadata = indicesIterator.next();
+            var settings = indexMetadata.getSettings();
+            var subscriptionName = REPLICATION_SUBSCRIPTION_NAME.get(settings); // Can be null.
+            if (subscriptionToDrop.equals(subscriptionName)) {
+                relevantIndices.add(indexMetadata);
+            }
+        }
+        return relevantIndices;
+    }
+
+    private List<OpenCloseTable> getSubscriptionTables(@Nonnull String subscriptionToDrop, Schemas schemas) {
+        return InformationSchemaIterables.tablesStream(schemas)
```

Comment:
`OpenCloseTable` reflects what we have in `OpenCloseRequest` - `RelationName` and `partitionIndexName`. But I pass null as partition index - because in close/open I'm interested in all partitions.

Is passing null correct in that sense? If so, remove `OpenCloseTable` class at all (just use List<RelationName>)
and also remove `List<OpenCloseTable> getSubscriptionTables` and use subscription.relations() once #11958 is merged.